File: fixtures/typescript/repomap/email.ts
import { formatDate } from './utils';
import { logEvent } from './logging';
import type { User } from './models';

export function sendWelcomeEmail(user: User): boolean {
  const sentOn = formatDate(new Date());
  logEvent(`welcome email queued for ${user.email} on ${sentOn}`);
  return true;
}

export function sendPasswordReset(user: User): boolean {
  logEvent(`password reset email queued for ${user.email}`);
  return true;
}

export function renderTemplate(name: string, data: unknown): string {
  return `<template ${name}>${JSON.stringify(data)}</template>`;
}

File: fixtures/python/repomap/email.py
import datetime

from logging import log_event
from utils import format_date


def send_welcome_email(user):
    sent_on = format_date(datetime.datetime.now())
    log_event(f"welcome email queued for {user['email']} on {sent_on}")
    return True


def send_password_reset(user):
    log_event(f"password reset email queued for {user['email']}")
    return True


def render_template(name, data):
    return f"<template {name}>{data}</template>"
